.ci/package-storage-publish.groovy

```diff
@@ -51,8 +51,8 @@ pipeline {
     stage('Build package') {
       steps {
         cleanup()
-        withGoEnv() {
-          dir("${BASE_DIR}") {
+        dir("${BASE_DIR}") {
+          withGoEnv() {
```

Comment:
Needed to be able to find the `.go-version` file in this repository, or it would use the default version 1.18.8.

https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Felastic-package-package-storage-publish/detail/PR-1046/1/pipeline/

Version 1.18.8 does not have the required method to be used here.